src/main/scala/is/hail/variant/Variant.scala

```diff
@@ -258,21 +243,21 @@ object VariantSubgen {
 }
 
 case class VariantSubgen(
-  contigGen: Gen[Contig],
+  contigGen: Gen[(String, Int)],
   nAllelesGen: Gen[Int],
   refGen: Gen[String],
   altGen: Gen[String]) {
 
   def gen: Gen[Variant] =
-    for (contig <- contigGen;
-      start <- Gen.choose(1, contig.length);
+    for ((contig, length) <- contigGen;
+      start <- Gen.choose(1, length);
       nAlleles <- nAllelesGen;
       ref <- refGen;
       altAlleles <- Gen.distinctBuildableOfN[Array, String](
         nAlleles,
         altGen)
         .filter(!_.contains(ref))) yield
-      Variant(contig.name, start, ref, altAlleles.tail.map(alt => AltAllele(ref, alt)))
+      Variant(contig, start, ref, altAlleles.tail.map(alt => AltAllele(ref, alt)))
```

Comment:
Why are you taking the tail here rather than using `nAlleles - 1` in `altAlleles`?